packages/hoppscotch-common/src/services/persistence/validation-schemas/index.ts

```diff
@@ -0,0 +1,444 @@
+import {
+  Environment,
+  GQLHeader,
+  HoppGQLAuth,
+  HoppGQLRequest,
+  HoppRESTRequest,
+} from "@hoppscotch/data"
+import { entityReference } from "verzod"
+import { z } from "zod"
+
+const ThemeColorSchema = z.enum([
+  "green",
+  "teal",
+  "blue",
+  "indigo",
+  "purple",
+  "yellow",
+  "orange",
+  "red",
+  "pink",
+])
+
+const BgColorSchema = z.enum(["system", "light", "dark", "black"])
+
+export const SettingsDefSchema = z.object({
+  syncCollections: z.boolean(),
+  syncHistory: z.boolean(),
+  syncEnvironments: z.boolean(),
+  PROXY_URL: z.string(),
+  CURRENT_INTERCEPTOR_ID: z.string(),
+  URL_EXCLUDES: z.object({
+    auth: z.boolean(),
+    httpUser: z.boolean(),
+    httpPassword: z.boolean(),
+    bearerToken: z.boolean(),
+    oauth2Token: z.boolean(),
+  }),
+  THEME_COLOR: ThemeColorSchema,
+  BG_COLOR: BgColorSchema,
+  TELEMETRY_ENABLED: z.boolean(),
+  EXPAND_NAVIGATION: z.boolean(),
+  SIDEBAR: z.boolean(),
+  SIDEBAR_ON_LEFT: z.boolean(),
+  COLUMN_LAYOUT: z.boolean(),
+})
+
+// @ts-expect-error recursive schema
+const HoppCollectionSchema = z.object({
+  v: z.number(),
+  name: z.string(),
+  folders: z.array(z.lazy(() => HoppCollectionSchema)),
+  id: z.optional(z.string()),
+})
+
+const HoppRESTRequestSchema = entityReference(HoppRESTRequest)
+
+const HoppGQLRequestSchema = entityReference(HoppGQLRequest)
+
+const HoppRESTCollectionSchema = HoppCollectionSchema.merge(
+  z.object({
+    requests: z.optional(z.array(HoppRESTRequestSchema)),
+  })
+)
+
+const HoppGQLCollectionSchema = HoppCollectionSchema.merge(
+  z.object({
+    requests: z.optional(z.array(HoppGQLRequestSchema)),
+  })
+)
+export const VUEX_SCHEMA = z.object({
+  postwoman: z.optional(
+    z.object({
+      settings: z.optional(SettingsDefSchema),
+      //! Versioned entities
+      collections: z.optional(z.array(HoppRESTCollectionSchema)),
+      collectionsGraphql: z.optional(z.array(HoppGQLCollectionSchema)),
+      environments: z.optional(z.array(entityReference(Environment))),
+    })
+  ),
+})
```

Comment:
There wasn't a well-defined type for the data associated with the key `vuex` read from `localStorage`. I've made the schema not strict and accept unknown keys, appreciate suggestions.

Reply:
Yep, the old schema was very iffy, most of our users should have migrated over to the new one, so we just need to worry about the existing legacy conversions tbh.